### PR TITLE
feat!: getIdentitiesContractKeys endpoint

### PR DESCRIPTION
**Pull request by markin-io**
## Issue being fixed or feature implemented

Remove `getIdentities` and `getIdentitiesByPublicKeyHashes` queries, and add `getIdentitiesContractKeys` as a replacement

## What was done?

- Remove queries from proto messages
- Remove queries from drive
- Update JS and Rust clients 
- Added `getIdentitiesContractKeys` and e2e test

## How Has This Been Tested?

- CI

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have added "!" to the title and described breaking changes in the corresponding section if my code contains any
- [ ] I have made corresponding changes to the documentation if needed

**For repository code-owners and collaborators only**
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

File: packages/platform-test-suite/test/functional/platform/proofs.spec.js
Comment (shumkov) [inline packages/platform-test-suite/test/functional/platform/proofs.spec.js:236]: Just remove?

**Review by shumkov — CHANGES_REQUESTED** [review]

**Review by shumkov — DISMISSED** [review]
👍

File: packages/rs-drive-abci/src/query/identity_based_queries/identities/mod.rs
Comment (QuantumExplorer) [inline packages/rs-drive-abci/src/query/identity_based_queries/identities/mod.rs:31]: You forgot to remove the query version.

**Review by QuantumExplorer — CHANGES_REQUESTED** [review]

**Review by shumkov — APPROVED** [review]
👍

**Comment by QuantumExplorer** [comment]
Let's wait on the new methods needed before this one is merged in.

File: packages/dapi-grpc/protos/platform/v0/platform.proto
Comment (QuantumExplorer) [inline packages/dapi-grpc/protos/platform/v0/platform.proto:242]: How is this going to work if you don't have the SearchKey?

File: packages/dapi-grpc/protos/platform/v0/platform.proto
Comment (QuantumExplorer) [inline packages/dapi-grpc/protos/platform/v0/platform.proto:255]: We don't use a map anywhere else and instead give back the values as a vec in the same order it was requested. I'm not super against using a map, I just think that it breaks consistency.

File: packages/dapi-grpc/build.rs
Comment (QuantumExplorer) [inline packages/dapi-grpc/build.rs:69]: Same

File: packages/rs-drive-abci/src/query/identity_based_queries/identities_keys/v0/mod.rs
Comment (QuantumExplorer) [inline packages/rs-drive-abci/src/query/identity_based_queries/identities_keys/v0/mod.rs:49]: todo...

**Review by QuantumExplorer — CHANGES_REQUESTED** [review]
The reason why we are not querying all identity information is because we could have 100s of keys on each identity. There is no benefit in getting back all keys for many identities. The goal here is to get specific keys for many identities. For example in Dashpay Dapp if I had 25 contacts, I would want to query the Dashpay contract bound encryption and decryption keys for the 25 contacts I have.

**Review by QuantumExplorer — CHANGES_REQUESTED** [review]
The reason why we are not querying all identity information is because we could have 100s of keys on each identity. There is no benefit in getting back all keys for many identities. The goal here is to get specific keys for many identities. For example in Dashpay Dapp if I had 25 contacts, I would want to query the Dashpay contract bound encryption and decryption keys for the 25 contacts I have.

**Review by QuantumExplorer — DISMISSED** [review]

**Review by QuantumExplorer — APPROVED** [review]

**Pull request merged**
